File: 22 November 2022/Python/Magic Triplets.py
#User function Template for python3

class Solution:
	def countTriplets(self, nums):
        ans=0
        for i in range(len(nums)):
           left=0
           right=0
           for j in range(i-1,-1,-1):
               if nums[i]>nums[j]:
                   left+=1
           for j in range(i+1,len(nums)):
               if nums[i]<nums[j]:
                   right+=1
           ans+=left*right
        return ans

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
	T=int(input())
	for i in range(T):
		n = int(input())
		nums = list(map(int,input().split()))
		ob = Solution()
		ans = ob.countTriplets(nums)
		print(ans)

# } Driver Code Ends

File: 22 November 2022/C++/Magic Triplets.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution{
	public:
	int countTriplets(vector<int>nums){
		int ans = 0;
	    for(int i = 1 ; i < nums.size()-1 ; i++)
	    {
	        int left = 0, right = 0;
	        for(int j = i-1 ; j >= 0 ; j--)
	        {
	            if(nums[i] > nums[j])left++;
	        }
	        
	        for(int j = i + 1 ; j < nums.size() ; j++)
	        {
	            if(nums[i] < nums[j])right++;
	        }
	        ans+=(left*right);
	    }
	    return ans;
	}
};

//{ Driver Code Starts.
int main(){
	int tc;
	cin >> tc;
	while(tc--){
		int n;
		cin >> n;
		vector<int>nums(n);
		for(int i = 0; i < n; i++)cin >> nums[i];
		Solution ob;
		int ans = ob.countTriplets(nums);
		cout << ans << "\n";
	}  
	return 0;
}
// } Driver Code Ends